packages/notifier/src/notifier.js

```diff
@@ -0,0 +1,54 @@
+import { producePromise } from '@agoric/produce-promise';
+import harden from '@agoric/harden';
+
+/**
+ * Makes a notifier object, which allows a service to produce a stream of update
+ * promises. Each update is a record containing
+ *      { currentState, nextStatePromise, updateHandle }.
+ *   currentState is whatever state the service wants to publish,
+ *   nextStatePromise is a HandledPromise, which resolves when the state changes
+ *   updateHandle is an opaque object that identifies the update.
+ *
+ * The notifier has two methods: getUpdateSince(handle), and updateState(state).
+ * getUpdateSince() returns the record above, or a Promise for it. If the
+ * argument to getUpdateSince(handle) is omitted or differs from the current
+ * handle, the current record is returned. If the handle matches the current
+ * state, then a promise is returned, which will resolve to the next record
+ * after the state next changes. nextStatePromise in the record is that same
+ * promise.
```

Comment:
Are you sure? I read the code carefully and believe it actually implements the more intuitive behavior, where the stateRecord contains a promise for the next state record.

Would any of the existing test cases reveal what it actually does?

Reply:
Your code has a clean and documentable behavior for closeStream, where undefined appears where the next state record was expected. However, rather than have the state record actually hold an undefined, I think it's better if it holds a promise fulfilled to undefined, since users need to prepare for that case anyway. This reduces the state space users need to test.

Reply:
I think I'm wrong about closing the stream and your code is correct.

Reply:
Thanks. I had a small niggle in the back of my head about that inconsistency. I'm more comfortable with nextStatePromise always being a promise. 

Should it fulfill to undefined, or a record containing the state, a promise for itself, and no updateHandle?
```
  currentResponse = { currentState,  nextStatePromise: currentPromiseRec.promise, undefined };
  currentPromiseRec.resolve(currentResponse);
```
That's not quite right, since we want to be able to discard the notifier after close.

Hmm. Let's talk.